src/glutil.cpp

```diff
@@ -32,61 +47,80 @@ void TransformState::setUniforms(GLuint prog) const
     setUniform(prog, "modelViewProjectionMatrix", mvproj);
 }
 
-
-void TransformState::load() const
+void TransformState::setOrthoProjection(double left, double right, double bottom, double top, double nearVal, double farVal)
 {
-    glMatrixMode(GL_PROJECTION);
-    glLoadMatrix(M44f(projMatrix));
-    glMatrixMode(GL_MODELVIEW);
-    glLoadMatrix(M44f(modelViewMatrix));
-}
+    // Note: Create an orthographic matrix from input values
+    if ((left == right) || (bottom == top) || (nearVal == farVal))
+        throw DisplazError("Invalid input for orthographic projection.");
+
+    double xx = 2.0/(right-left);
+    double yy = 2.0/(top-bottom);
+    double zz = -2.0/(farVal-nearVal);
+
+    double tx = -(right+left)/(right-left);
+    double ty = -(top+bottom)/(top-bottom);
+    double tz = -(farVal+nearVal)/(farVal-nearVal);
 
+    projMatrix = M44d(xx,0.0,0.0,0.0,
+                      0.0,yy,0.0,0.0,
+                      0.0,0.0,zz,0.0,
+                      tx,ty,tz,1.0);
+}
 
 //------------------------------------------------------------------------------
 
-void drawBoundingBox(const TransformState& transState,
-                     const Imath::Box3d& bbox,
-                     const Imath::C3f& col)
+void drawBox(const TransformState& transState,
+             const Imath::Box3d& bbox,
+             const Imath::C3f& col,
+             const GLuint& shaderProgram)
 {
     // Transform to box min for stability with large offsets
+    // This function is allowing you to render any box, not necessarily having anything to do
+    // with the bounding box (e.g. hcloudview is using this in drawBounds)
     TransformState trans2 = transState.translate(bbox.min);
     Imath::Box3f box2(V3f(0), V3f(bbox.max - bbox.min));
-    drawBoundingBox(trans2, box2, col);
+    drawBox(trans2, box2, col, shaderProgram);
 }
 
+void drawBox(const TransformState& transState,
+             const Imath::Box3f& bbox,
+             const Imath::C3f& col,
+             const GLuint& shaderProgram)
+{
+    // This function is allowing you to render any box, not necessarily having anything to do
+    // with the bounding box (e.g. hcloudview is using this in drawBounds)
+    // TODO: FIX ME
+    tfm::printfln("drawBox :: has not been implemented, yet.");
+}
 
 void drawBoundingBox(const TransformState& transState,
-                     const Imath::Box3f& bbox,
-                     const Imath::C3f& col)
+                     const GLuint& bboxVertexArray,
+                     const Imath::V3f& offset,
+                     const Imath::C3f& col,
+                     const GLuint& shaderProgram)
```

Comment:
This seems to have become something more like `draw12SmoothLines()` at this stage ;-)  Looks like I'll need to refactor the drawing code into a bunch of utilities... using core context only really is a different ballgame.

Thanks for all the work here, I've learned quite a bit about modern OpenGL while looking through it.